File: node/transcribe_file_short_separate_recognition.js
const { SpeechClient } = require("@soniox/soniox-node");

/**
 * Set your Soniox Cloud API key:
 * from command line: export SONIOX_API_KEY=<YOUR-API-KEY>
 * or
 * pass config object: new SpeechClient({ api_key: "<YOUR-API-KEY>" })
 */
const speechClient = new SpeechClient();

(async function () {
  const channel_results = await speechClient.transcribeFileShort(
    "../test_data/test_audio_multi_channel.flac",
    {
      num_audio_channels: 2,
      enable_separate_recognition_per_channel: true,
    }
  );

  for (const result of channel_results) {
    console.log(
      `Channel ${result.channel}: ${result.words
        .map((word) => word.text)
        .join(" ")}`
    );
  }
})();

File: python/transcribe_file_short_separate_recognition.py
from soniox.transcribe_file import transcribe_file_short
from soniox.speech_service import SpeechClient


def main():
    with SpeechClient() as client:
        channel_results = transcribe_file_short(
            "../test_data/test_audio_multi_channel.flac",
            client,
            num_audio_channels=2,
            enable_separate_recognition_per_channel=True,
        )
        for result in channel_results:
            print(f"Channel {result.channel}: " + " ".join(word.text for word in result.words))


if __name__ == "__main__":
    main()
